src/server/game/Spells/Spell.cpp

```diff
@@ -2936,6 +2938,27 @@ void Spell::prepare(SpellCastTargets const* targets, AuraEffect const* triggered
     else
         m_casttime = m_spellInfo->CalcCastTime(this);
 
+    if (m_caster->GetTypeId() == TYPEID_UNIT) // _UNIT actually means creature. for some reason.
```

Comment:
@ccrs Check Entities/Object/Object.h for the definition of WorldObject::ToCreature().